src/posix/platform/hdlc_interface.cpp

```diff
@@ -247,6 +247,94 @@ otError HdlcInterface::Write(const uint8_t *aFrame, uint16_t aLength)
     return error;
 }
 
+otError HdlcInterface::WaitResponse(struct timeval &aTimeout)
+{
+    otError error = OT_ERROR_NONE;
+
+#if OPENTHREAD_POSIX_VIRTUAL_TIME
+    struct Event event;
+
+    platformSimSendSleepEvent(&aTimeout);
+    platformSimReceiveEvent(&event);
+
+    switch (event.mEvent)
+    {
+    case OT_SIM_EVENT_RADIO_SPINEL_WRITE:
+        Decode(event.mData, event.mDataLength);
+        break;
+
+    case OT_SIM_EVENT_ALARM_FIRED:
+        ExitNow(error = OT_ERROR_RESPONSE_TIMEOUT);
+        break;
+
+    default:
+        assert(false);
+        break;
+    }
+#else // OPENTHREAD_POSIX_VIRTUAL_TIME
+    fd_set read_fds;
+    fd_set error_fds;
+    int rval;
+
+    FD_ZERO(&read_fds);
+    FD_ZERO(&error_fds);
+    FD_SET(mSockFd, &read_fds);
+    FD_SET(mSockFd, &error_fds);
+
+    rval = select(mSockFd + 1, &read_fds, NULL, &error_fds, &aTimeout);
+
+    if (rval > 0)
+    {
+        if (FD_ISSET(mSockFd, &read_fds))
+        {
+            Read();
```

Comment:
We can possibly get part of a full HDLC frame from `Read()` and in that case we return from `WaitResponse()` without any full spinel frame received.  

Checking how we use `WaitForResponse()` in the `RadioSpinel`, I think this behavior is fine, but we should be careful of this case.

I would suggest maybe changing the `WaitForResponse()` description.

Thoughts?

Reply:
For HDLC interface, it indeed may get part of a full HDLC frame from `read()`. I have updated the description for the function `WaitResponse()`.